engine/model/job.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"go.uber.org/zap"
+)
+
+type (
+	// JobID is the unique identifier of a job
+	JobID = string
+	// JobType is the type of a type, like `CVSDEMO`, `DM`, `CDC`.
+	JobType = int32
+)
+
+// Define job type
+const (
+	JobTypeInvalid = JobType(iota)
+	// job type for job manager, only for inner use
+	JobTypeJobManager
+	JobTypeCVSDemo
+	JobTypeDM
+	JobTypeCDC
+	JobTypeFakeJob
+)
+
+var jobTypeName = map[int32]string{
+	2: "CVSDemo",
+	3: "DM",
+	4: "CDC",
+	5: "FakeJob",
+}
+
+var jobTypeValue = map[string]int32{
+	"CVSDemo": 2,
+	"DM":      3,
+	"CDC":     4,
+	"FakeJob": 5,
+}
+
+// MustJobTypeByName get job type by name. Panic if it fails.
+func MustJobTypeByName(name string) JobType {
+	tp, ok := JobTypeByName(name)
+	if !ok {
+		log.L().Panic("unexpected fail when get job type by name", zap.String("job_name", name))
+	}
+
+	return tp
+}
+
+// JobTypeByName get JobType by readable job name
+func JobTypeByName(name string) (JobType, bool) {
+	tp, exists := jobTypeValue[name]
+	if !exists {
+		return JobTypeInvalid, false
+	}
+
+	return tp, true
+}
+
+// MustJobNameByType get job name by job type. Panic if it fails.
+func MustJobNameByType(tp JobType) string {
+	name, ok := JobNameByType(tp)
+	if !ok {
+		log.L().Panic("unexpected fail when get job name by type", zap.Int32("job_type", tp))
+	}
+
+	return name
+}
+
+// JobNameByType get job name by job type
+func JobNameByType(tp JobType) (string, bool) {
```

Comment:
How about implementing `fmt.Stringer()`. I think it would be idiomatic.

Reply:
This function has a meaning to check if the job type is registered.
`fmt.Stringer()` seems not so precise。

Reply:
This purpose would make me think the string is actually the job type identifier. To check whether a type is valid, it's unnecessary to get its string name. In most cases, I think the integer value is the only identifier and the string is just for human-readable.

BTW, here we should use `JobTypeName` instead.

Reply:
Besides, `fmt.Stringer()` is more convenient for log and debug.

Reply:
I will think about it. Seems it's ambiguous for the name.